sympy/core/expr.py

```diff
@@ -3670,7 +3661,7 @@ def round(self, n=None):
             return x
         if not x.is_extended_real:
             i, r = x.as_real_imag()
-            return i.round(n) + S.ImaginaryUnit*r.round(n)
+            return round(n) + S.ImaginaryUnit*round(n)
```

Comment:
I'm not sure if this is required.
Anyway, it should be something like:

```suggestion
            return round(i, n) + S.ImaginaryUnit*round(r, n)
```

Reply:
It looks like only `ComplexInfinity` and `NaN` have `is_extended_real` not truthy
and those are handled by the clause above so I'm removing this code.

I left an assert to double check during the tests but it can also be removed (if someone says Eyyy)